content/blog/multitenancy-for-kubernetes-clusters-using-hpe-alletra-6000-and-nimble-storage.md

```diff
@@ -0,0 +1,235 @@
+
+With the introduction of [HPE CSI Driver for Kubernetes 2.0](https://community.hpe.com/t5/Around-the-Storage-Block/HPE-CSI-Driver-for-Kubernetes-now-available-for-HPE-Alletra/ba-p/7136280) and the software powering HPE Alletra 6000 and Nimble Storage, Hewlett Packard Enterprise introduces multitenancy for Kubernetes clusters accessing persistent volumes on the aforementioned storage arrays.
+
+In this blog post we'll step through some of the basic elements to enable storage administrators to safely hand over credentials to Kubernetes administrators.
+
+# The enabling primitives
+
+HPE Alletra 6000 and NimbleOS 6.0 includes a new command-line interface (CLI) called `tenantadmin`. This new CLI enables storage administrators to confine a tenant user account into specific folders. Folders need to exist on the array prior creating a new tenant.
+
+The synopsis of the `--help` help flag gives an overview of the supported workflows.
+
+```markdown
+HPE Alletra $ tenantadmin --help
+Usage: tenantadmin [options]
+Manage Tenants.
+
+Available options are:
+  --help                           Program help.
+
+  --list                           List Tenants.
+
+  --info name                      Tenant info.
+
+  --add tenant_name                Add a tenant.
+    --folders folders              List of folder paths (comma separated
+                                   pool_name:fqn) the tenant will be able to
+                                   access (mandatory).
+
+  --remove name                    Remove a tenant.
+
+  --add_folder tenant_name         Add a folder path for tenant access.
+    --name folder_name             Name of the folder path (pool_name:fqn) to
+                                   be added (mandatory).
+
+  --remove_folder tenant_name      Remove a folder path from tenant access.
+    --name folder_name             Name of the folder path (pool_name:fqn) to
+                                   be removed (mandatory).
+
+  --passwd                         Change tenant's login password.
+    --tenant name                  Change a specific tenant's login password
+                                   (mandatory).
+```
+
+With these basic create, read, update and delete (CRUD) elements, the storage administrator is now empowered to delegate and confine all the storage resource management to a folder for a Kubernetes administrator to household with.
+
+# The storage administrator's workflow
+
+Assume a new Kubernetes environment is being deployed within an Enterprise. Each cluster needs to be compartmentalized to not consume all the performance and capacity of the array.
+
+First step, create a new restricted folder in the pool of your choice.
+
+```markdown
+folder --create k8s-prod \
+  --iops_limit 75000 --usage_limit 2500000 \
+  --description="Kubernetes Production Cluster"
+```
+
+Next, create a new tenant. 
```

Comment:
Is a tenant a user or a group of users. Maybe you may want to define what is a tenant in the context of HPE Alletra and Nimble Storage. The output of the command tenantadmin says "Created User K8sAdminProd". So I assume a Tenant is a User in this context. But as a user of Ezmeral Container Platform, a tenant is a group of users.